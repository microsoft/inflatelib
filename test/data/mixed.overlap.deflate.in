
# This writes "This tests overlapped writes" as uncompressed data and then copies it the following number of times, each
# using dynamic and static tables
#   * Repeat the string 9 times, offset at the beginning of the string (overlapping 8 times): (252, 28)
#   * Repeat the string 9 times, offset 4 strings back (overlapping 5 times): (252, 112)
#   * Repeat the string 9 times, offset 8 strings back (overlapping once): (252, 224)
# NOTE: This test is only applicable to Deflate

# First block: uncompressed
>1
0       # bfinal = false
00      # No compression

>16
001C    # LEN = 1344
FFE3    # NLEN = ~LEN

>8
"This tests overlapped writes"

# Now with dynamic compression
>1
0       # bfinal = false
10      # Dynamic compression

11100   # HLIT = 285 (28 + 257)
01111   # HDIST = 16 (15 + 1)
1110    # HCLEN = 18 (14 + 4)

# Code Length Alphabet Code Lengths:
000 011 010 011 000 000 000 000 000 000 000 000 000 000 000 010 000 010

# Literal/Length & Distance Alphabet Code Lengths:
>>1
10 >1 1111111 >>1
10 >1 1101011 >>1
00
10 >1 0010000 >>1
00
111 >1 110 >>1
00
111 >1 000 >>1
01
110
01

# Literal/Length Tree:
#   Symbol      Bit Count   Code
#   END         1           0
#   284         1           1
#
# Distance Tree:
#   Symbol      Bit Count   Code
#   9           1           0
#   13          2           10
#   15          2           11

# Encoded Data:
>>1
1 >1 11001 >>1 0 >1 011 >>1
1 >1 11001 >>1 10 >1 01111 >>1
1 >1 11001 >>1 11 >1 011111 >>1
0

# And finally, with static compression
>1
1       # bfinal = true
01      # Static compression

# Encoded Data:
>>1
11000100 >1 11001 >>1 01001 >1 011 >>1
11000100 >1 11001 >>1 01101 >1 01111 >>1
11000100 >1 11001 >>1 01111 >1 011111 >>1
0000000
