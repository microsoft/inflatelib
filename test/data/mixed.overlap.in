
# Repeats "London Bridge" (1344 in length) many times using all three encodings such that each one "overflows" the window

# First with no compression. We want to repeat the string 49 times in total to give a total of 65,856 bytes, however
# uncompressed blocks have a max length of 65,535, so we need to do this in two steps.
>1
0       # bfinal = false
00      # No compression

# First instance, just write the string once
>16
0540    # LEN = 1344
FABF    # NLEN = ~LEN

>8
"London Bridge is falling down,\n"
"Falling down, falling down,\n"
"London Bridge is falling down,\n"
"My fair Lady.\n"
"\n"
"Build it up with wood and clay,\n"
"Wood and clay, wood and clay,\n"
"Build it up with wood and clay,\n"
"My fair Lady.\n"
"\n"
"Wood and clay will wash away,\n"
"Wash away, wash away,\n"
"Wood and clay will wash away,\n"
"My fair Lady.\n"
"\n"
"Build it up with bricks and mortar,\n"
"Bricks and mortar, bricks and mortar,\n"
"Build it up with bricks and mortar,\n"
"My fair Lady.\n"
"\n"
"Bricks and mortar will not stay,\n"
"Will not stay, will not stay,\n"
"Bricks and mortar will not stay,\n"
"My fair Lady.\n"
"\n"
"Build it up with iron and steel,\n"
"Iron and steel, iron and steel,\n"
"Build it up with iron and steel,\n"
"My fair Lady.\n"
"\n"
"Iron and steel will bend and bow,\n"
"Bend and bow, bend and bow,\n"
"Iron and steel will bend and bow,\n"
"My fair Lady.\n"
"\n"
"Build it up with silver and gold,\n"
"Silver and gold, silver and gold,\n"
"Build it up with silver and gold,\n"
"My fair Lady.\n"
"\n"
"Silver and gold will be stolen away,\n"
"Stolen away, stolen away,\n"
"Silver and gold will be stolen away,\n"
"My fair Lady.\n"
"\n"
"Set a man to watch all nigh,\n"
"Watch all night, watch all night,\n"
"Set a man to watch all night,\n"
"My fair Lady.\n"
"\n"
"Suppose the man should fall asleep,\n"
"Fall asleep, fall asleep,\n"
"Suppose the man should fall asleep?\n"
"My fair Lady.\n"
"\n"
"Give him a pipe to smoke all night,\n"
"Smoke all night, smoke all night,\n"
"Give him a pipe to smoke all night,\n"
"My fair Lady.\n"

>1
0       # bfinal = false
00      # No compression

# Second instance, just write the string 48 times (64,512 bytes)
>16
FC00    # LEN = 1344
03FF    # NLEN = ~LEN

>8
repeat (48) {
    "London Bridge is falling down,\n"
    "Falling down, falling down,\n"
    "London Bridge is falling down,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with wood and clay,\n"
    "Wood and clay, wood and clay,\n"
    "Build it up with wood and clay,\n"
    "My fair Lady.\n"
    "\n"
    "Wood and clay will wash away,\n"
    "Wash away, wash away,\n"
    "Wood and clay will wash away,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with bricks and mortar,\n"
    "Bricks and mortar, bricks and mortar,\n"
    "Build it up with bricks and mortar,\n"
    "My fair Lady.\n"
    "\n"
    "Bricks and mortar will not stay,\n"
    "Will not stay, will not stay,\n"
    "Bricks and mortar will not stay,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with iron and steel,\n"
    "Iron and steel, iron and steel,\n"
    "Build it up with iron and steel,\n"
    "My fair Lady.\n"
    "\n"
    "Iron and steel will bend and bow,\n"
    "Bend and bow, bend and bow,\n"
    "Iron and steel will bend and bow,\n"
    "My fair Lady.\n"
    "\n"
    "Build it up with silver and gold,\n"
    "Silver and gold, silver and gold,\n"
    "Build it up with silver and gold,\n"
    "My fair Lady.\n"
    "\n"
    "Silver and gold will be stolen away,\n"
    "Stolen away, stolen away,\n"
    "Silver and gold will be stolen away,\n"
    "My fair Lady.\n"
    "\n"
    "Set a man to watch all nigh,\n"
    "Watch all night, watch all night,\n"
    "Set a man to watch all night,\n"
    "My fair Lady.\n"
    "\n"
    "Suppose the man should fall asleep,\n"
    "Fall asleep, fall asleep,\n"
    "Suppose the man should fall asleep?\n"
    "My fair Lady.\n"
    "\n"
    "Give him a pipe to smoke all night,\n"
    "Smoke all night, smoke all night,\n"
    "Give him a pipe to smoke all night,\n"
    "My fair Lady.\n"
}

# Now with dynamic compression. Write the string once before repeating it 48 times using a distance of 2688 (2 strings)
>1
0       # bfinal = false
10      # Dynamic compression

11101   # HLIT = 286 (29 + 257)
10110   # HDIST = 23 (22 + 1)
1110    # HCLEN = 18 (14 + 4)

# Code Length Alphabet Code Lengths:
000 100 011 011 101 011 100 011 101 011 000 011 000 101 000 000 000 101

# Literal/Length & Distance Alphabet Code Lengths:
>>1
1101 >1 111 >>1
010
101 >1 0001010 >>1
11101
101 >1 0000000 >>1
010
000
100
101 >1 0000101 >>1
1100
000
000
100
1101 >1 000 >>1
1100
1100
000
11110
000
000
100
100
1101 >1 010 >>1
100
1101 >1 000 >>1
11110
1101 >1 110 >>1
11101
100
011
001
010
011
011
011
001
000
100
001
011
001
001
011
000
010
010
010
011
100
010
000
010
101 >1 1111011 >>1
11111
101 >1 0010001 >>1
11111
101 >1 0001011 >>1
11100

# Literal/Length Tree:
#   Symbol      Bit Count   Code
#   10          5           10010
#   ' '         3           000
#   ','         5           10011
#   '.'         7           1110110
#   '?'         9           111111100
#   'B'         7           1110111
#   'F'         9           111111101
#   'G'         9           111111110
#   'I'         8           11111100
#   'L'         7           1111000
#   'M'         7           1111001
#   'S'         7           1111010
#   'W'         8           11111101
#   'a'         3           001
#   'b'         7           1111011
#   'c'         6           110100
#   'd'         4           0100
#   'e'         5           10100
#   'f'         6           110101
#   'g'         6           110110
#   'h'         6           110111
#   'i'         4           0101
#   'k'         7           1111100
#   'l'         4           0110
#   'm'         6           111000
#   'n'         4           0111
#   'o'         4           1000
#   'p'         6           111001
#   'r'         5           10101
#   's'         5           10110
#   't'         5           10111
#   'u'         6           111010
#   'v'         7           1111101
#   'w'         5           11000
#   'y'         5           11001
#   END         10          1111111110
#   285         10          1111111111
#
# Distance Tree:
#   Symbol      Bit Count   Code
#   22          1           0

# Encoded Data:
>>1
1111000 1000 0111 0100 1000 0111 000 1110111 10101 0101 0100 110110 10100 000 0101 10110 000 110101 001 0110 0110 0101 0111 110110 000 0100 1000 11000 0111 10011 10010
111111101 001 0110 0110 0101 0111 110110 000 0100 1000 11000 0111 10011 000 110101 001 0110 0110 0101 0111 110110 000 0100 1000 11000 0111 10011 10010
1111000 1000 0111 0100 1000 0111 000 1110111 10101 0101 0100 110110 10100 000 0101 10110 000 110101 001 0110 0110 0101 0111 110110 000 0100 1000 11000 0111 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 11000 1000 1000 0100 000 001 0111 0100 000 110100 0110 001 11001 10011 10010
11111101 1000 1000 0100 000 001 0111 0100 000 110100 0110 001 11001 10011 000 11000 1000 1000 0100 000 001 0111 0100 000 110100 0110 001 11001 10011 10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 11000 1000 1000 0100 000 001 0111 0100 000 110100 0110 001 11001 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
11111101 1000 1000 0100 000 001 0111 0100 000 110100 0110 001 11001 000 11000 0101 0110 0110 000 11000 001 10110 110111 000 001 11000 001 11001 10011 10010
11111101 001 10110 110111 000 001 11000 001 11001 10011 000 11000 001 10110 110111 000 001 11000 001 11001 10011 10010
11111101 1000 1000 0100 000 001 0111 0100 000 110100 0110 001 11001 000 11000 0101 0110 0110 000 11000 001 10110 110111 000 001 11000 001 11001 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 1111011 10101 0101 110100 1111100 10110 000 001 0111 0100 000 111000 1000 10101 10111 001 10101 10011 10010
1110111 10101 0101 110100 1111100 10110 000 001 0111 0100 000 111000 1000 10101 10111 001 10101 10011 000 1111011 10101 0101 110100 1111100 10110 000 001 0111 0100 000 111000 1000 10101 10111 001 10101 10011 10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 1111011 10101 0101 110100 1111100 10110 000 001 0111 0100 000 111000 1000 10101 10111 001 10101 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1110111 10101 0101 110100 1111100 10110 000 001 0111 0100 000 111000 1000 10101 10111 001 10101 000 11000 0101 0110 0110 000 0111 1000 10111 000 10110 10111 001 11001 10011 10010
11111101 0101 0110 0110 000 0111 1000 10111 000 10110 10111 001 11001 10011 000 11000 0101 0110 0110 000 0111 1000 10111 000 10110 10111 001 11001 10011 10010
1110111 10101 0101 110100 1111100 10110 000 001 0111 0100 000 111000 1000 10101 10111 001 10101 000 11000 0101 0110 0110 000 0111 1000 10111 000 10110 10111 001 11001 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 0101 10101 1000 0111 000 001 0111 0100 000 10110 10111 10100 10100 0110 10011 10010
11111100 10101 1000 0111 000 001 0111 0100 000 10110 10111 10100 10100 0110 10011 000 0101 10101 1000 0111 000 001 0111 0100 000 10110 10111 10100 10100 0110 10011 10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 0101 10101 1000 0111 000 001 0111 0100 000 10110 10111 10100 10100 0110 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
11111100 10101 1000 0111 000 001 0111 0100 000 10110 10111 10100 10100 0110 000 11000 0101 0110 0110 000 1111011 10100 0111 0100 000 001 0111 0100 000 1111011 1000 11000 10011 10010
1110111 10100 0111 0100 000 001 0111 0100 000 1111011 1000 11000 10011 000 1111011 10100 0111 0100 000 001 0111 0100 000 1111011 1000 11000 10011 10010
11111100 10101 1000 0111 000 001 0111 0100 000 10110 10111 10100 10100 0110 000 11000 0101 0110 0110 000 1111011 10100 0111 0100 000 001 0111 0100 000 1111011 1000 11000 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 10110 0101 0110 1111101 10100 10101 000 001 0111 0100 000 110110 1000 0110 0100 10011 10010
1111010 0101 0110 1111101 10100 10101 000 001 0111 0100 000 110110 1000 0110 0100 10011 000 10110 0101 0110 1111101 10100 10101 000 001 0111 0100 000 110110 1000 0110 0100 10011 10010
1110111 111010 0101 0110 0100 000 0101 10111 000 111010 111001 000 11000 0101 10111 110111 000 10110 0101 0110 1111101 10100 10101 000 001 0111 0100 000 110110 1000 0110 0100 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1111010 0101 0110 1111101 10100 10101 000 001 0111 0100 000 110110 1000 0110 0100 000 11000 0101 0110 0110 000 1111011 10100 000 10110 10111 1000 0110 10100 0111 000 001 11000 001 11001 10011 10010
1111010 10111 1000 0110 10100 0111 000 001 11000 001 11001 10011 000 10110 10111 1000 0110 10100 0111 000 001 11000 001 11001 10011 10010
1111010 0101 0110 1111101 10100 10101 000 001 0111 0100 000 110110 1000 0110 0100 000 11000 0101 0110 0110 000 1111011 10100 000 10110 10111 1000 0110 10100 0111 000 001 11000 001 11001 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1111010 10100 10111 000 001 000 111000 001 0111 000 10111 1000 000 11000 001 10111 110100 110111 000 001 0110 0110 000 0111 0101 110110 110111 10011 10010
11111101 001 10111 110100 110111 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 000 11000 001 10111 110100 110111 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 10010
1111010 10100 10111 000 001 000 111000 001 0111 000 10111 1000 000 11000 001 10111 110100 110111 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
1111010 111010 111001 111001 1000 10110 10100 000 10111 110111 10100 000 111000 001 0111 000 10110 110111 1000 111010 0110 0100 000 110101 001 0110 0110 000 001 10110 0110 10100 10100 111001 10011 10010
111111101 001 0110 0110 000 001 10110 0110 10100 10100 111001 10011 000 110101 001 0110 0110 000 001 10110 0110 10100 10100 111001 10011 10010
1111010 111010 111001 111001 1000 10110 10100 000 10111 110111 10100 000 111000 001 0111 000 10110 110111 1000 111010 0110 0100 000 110101 001 0110 0110 000 001 10110 0110 10100 10100 111001 111111100 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
10010
111111110 0101 1111101 10100 000 110111 0101 111000 000 001 000 111001 0101 111001 10100 000 10111 1000 000 10110 111000 1000 1111100 10100 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 10010
1111010 111000 1000 1111100 10100 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 000 10110 111000 1000 1111100 10100 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 10010
111111110 0101 1111101 10100 000 110111 0101 111000 000 001 000 111001 0101 111001 10100 000 10111 1000 000 10110 111000 1000 1111100 10100 000 001 0110 0110 000 0111 0101 110110 110111 10111 10011 10010
1111001 11001 000 110101 001 0101 10101 000 1111000 001 0100 11001 1110110 10010
1111111111 >1 1111101111111101 >>1 0 >1 1001111111 >>1
1111111110

# Now with static compression. Write the string once before repeating it 48 times using a distance of 2688 (2 strings),
# same as we did with static compression
>1
1       # bfinal = true
01      # Static compression

# Encoded Data:
>>1
01111100 10011111 10011110 10010100 10011111 10011110 01010000 01110010 10100010 10011001 10010100 10010111 10010101 01010000 10011001 10100011 01010000 10010110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 00111010
01110110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 01010000 10010110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 00111010
01111100 10011111 10011110 10010100 10011111 10011110 01010000 01110010 10100010 10011001 10010100 10010111 10010101 01010000 10011001 10100011 01010000 10010110 10010001 10011100 10011100 10011001 10011110 10010111 01010000 10010100 10011111 10100111 10011110 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 00111010
10000111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 01010000 10100111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01010000 10100111 10011001 10011100 10011100 01010000 10100111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 01010000 10100111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000111 10011111 10011111 10010100 01010000 10010001 10011110 10010100 01010000 10010011 10011100 10010001 10101001 01010000 10100111 10011001 10011100 10011100 01010000 10100111 10010001 10100011 10011000 01010000 10010001 10100111 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10010010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 00111010
01110010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 01010000 10010010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10010010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01010000 10100111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 00111010
10000111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 01010000 10100111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 00111010
01110010 10100010 10011001 10010011 10011011 10100011 01010000 10010001 10011110 10010100 01010000 10011101 10011111 10100010 10100100 10010001 10100010 01010000 10100111 10011001 10011100 10011100 01010000 10011110 10011111 10100100 01010000 10100011 10100100 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10011001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 00111010
01111001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 01010000 10011001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10011001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01111001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 00111010
01110010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 01010000 10010010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 00111010
01111001 10100010 10011111 10011110 01010000 10010001 10011110 10010100 01010000 10100011 10100100 10010101 10010101 10011100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 10011110 10010100 01010000 10010001 10011110 10010100 01010000 10010010 10011111 10100111 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 00111010
10000011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 01010000 10100011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 00111010
01110010 10100101 10011001 10011100 10010100 01010000 10011001 10100100 01010000 10100101 10100000 01010000 10100111 10011001 10100100 10011000 01010000 10100011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 01010000 10100011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 01010000 10100011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 00111010
10000011 10011001 10011100 10100110 10010101 10100010 01010000 10010001 10011110 10010100 01010000 10010111 10011111 10011100 10010100 01010000 10100111 10011001 10011100 10011100 01010000 10010010 10010101 01010000 10100011 10100100 10011111 10011100 10010101 10011110 01010000 10010001 10100111 10010001 10101001 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000011 10010101 10100100 01010000 10010001 01010000 10011101 10010001 10011110 01010000 10100100 10011111 01010000 10100111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 01011100 00111010
10000111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 01010000 10100111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
10000011 10010101 10100100 01010000 10010001 01010000 10011101 10010001 10011110 01010000 10100100 10011111 01010000 10100111 10010001 10100100 10010011 10011000 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
10000011 10100101 10100000 10100000 10011111 10100011 10010101 01010000 10100100 10011000 10010101 01010000 10011101 10010001 10011110 01010000 10100011 10011000 10011111 10100101 10011100 10010100 01010000 10010110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01011100 00111010
01110110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01011100 01010000 10010110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01011100 00111010
10000011 10100101 10100000 10100000 10011111 10100011 10010101 01010000 10100100 10011000 10010101 01010000 10011101 10010001 10011110 01010000 10100011 10011000 10011111 10100101 10011100 10010100 01010000 10010110 10010001 10011100 10011100 01010000 10010001 10100011 10011100 10010101 10010101 10100000 01101111 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
00111010
01110111 10011001 10100110 10010101 01010000 10011000 10011001 10011101 01010000 10010001 01010000 10100000 10011001 10100000 10010101 01010000 10100100 10011111 01010000 10100011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
10000011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 01010000 10100011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
01110111 10011001 10100110 10010101 01010000 10011000 10011001 10011101 01010000 10010001 01010000 10100000 10011001 10100000 10010101 01010000 10100100 10011111 01010000 10100011 10011101 10011111 10011011 10010101 01010000 10010001 10011100 10011100 01010000 10011110 10011001 10010111 10011000 10100100 01011100 00111010
01111101 10101001 01010000 10010110 10010001 10011001 10100010 01010000 01111100 10010001 10010100 10101001 01011110 00111010
11000101 >1 1111101111111101 >>1 10110 >1 1001111111 >>1
0000000
