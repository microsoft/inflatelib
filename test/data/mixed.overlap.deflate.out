
repeat (55) {
    "This tests overlapped writes"
}
